distribution.pm

```diff
@@ -127,7 +129,8 @@ sub script_run ($self, $cmd, @args) {
         my $marker = "; echo $str-\$?-" . ($args{output} ? "Comment: $args{output}" : '');
         if (testapi::is_serial_terminal) {
             testapi::type_string($marker);
-            testapi::wait_serial($cmd . $marker, no_regex => 1, quiet => $args{quiet});
+            die 'Command was mistyped'
```

Comment:
I'm wondering if a `die` is really what we want.

Normally `script_run` returns undef if the wait_serial at the end returns undef.

It's the decision of the user of the function if they want it to be fatal (usually be using `assert_script_run`) or not.
OTOH a clear distinction that the typing itself didn't work vs. a non-zero return code from the script is a good idea.

If we decide we want this to die here, then maybe `autotest::croak` would make more sense, as that's also used in `_handle_script_run_ret` (called from `assert_script_run`).

Reply:
`croak` usually prints the call stack